x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+            const credentials = await resetCredentials(BASE_ENV_URL, environment.id, id, API_KEY);
+
+            // Base64 encode the credentials in order to invoke ES and KB APIs
+            const auth = encode(`${credentials.username}:${credentials.password}`);
+
+            // Wait for elasticsearch status to go green.
+            await waitForEsStatusGreen(environment.es_url, auth, id);
+
+            // Wait until Kibana is available
+            await waitForKibanaAvailable(environment.kb_url, auth, id);
+
+            // Normalized the set of available env vars in cypress
+            const cyCustomEnv = {
+              CYPRESS_BASE_URL: environment.kb_url,
+
+              ELASTICSEARCH_URL: environment.es_url,
+              ELASTICSEARCH_USERNAME: credentials.username,
+              ELASTICSEARCH_PASSWORD: credentials.password,
+
+              KIBANA_URL: environment.kb_url,
+              KIBANA_USERNAME: credentials.username,
+              KIBANA_PASSWORD: credentials.password,
+
+              CLOUD_SERVERLESS: true,
+            };
+
+            if (process.env.DEBUG && !process.env.CI) {
+              log.info(`
+              ----------------------------------------------
+              Cypress run ENV for file: ${filePath}:
+              ----------------------------------------------
+              ${JSON.stringify(cyCustomEnv, null, 2)}
+              ----------------------------------------------
+              `);
+            }
+
+            if (isOpen) {
+              await cypress.open({
+                configFile: cypressConfigFilePath,
+                config: {
+                  e2e: {
+                    baseUrl: environment.kb_url,
+                  },
+                  env: cyCustomEnv,
+                },
+              });
+            } else {
+              try {
+                result = await cypress.run({
+                  browser: 'electron',
+                  spec: filePath,
+                  configFile: cypressConfigFilePath,
+                  reporter: argv.reporter as string,
+                  reporterOptions: argv.reporterOptions,
+                  headed: argv.headed as boolean,
+                  config: {
+                    e2e: {
+                      baseUrl: environment.kb_url,
+                    },
+                    numTestsKeptInMemory: 0,
+                    env: cyCustomEnv,
+                  },
+                });
+              } catch (error) {
+                result = error;
+              }
+            }
+
+            // Delete serverless environment
+            await deleteEnvironment(
+              BASE_ENV_URL,
+              environment.id,
+              PROJECT_NAME,
+              id,
+              API_KEY,
+              onEarlyExit
+            );
+
+            return result;
+          });
+          return result;
+        },
+        {
+          concurrency: PARALLEL_COUNT,
+        }
+      ).then((results) => {
```

Comment:
```
const  results = pMap(...);
```

will help to get rid of `then`.

Reply:
Addressed in this commit: https://github.com/elastic/kibana/pull/169311/commits/1f9eb5acf38087e507038c77e95e955575acc2ce